Library/Compiler.php

```diff
@@ -1340,7 +1340,7 @@ public function processExtensionGlobals($namespace)
                 }
 
                 if (!isset($iniEntry['name'])) {
-                    $iniName = $name;
+                    $iniName = $namespace.'.'.$name;
```

Comment:
This is not fixes #1604. Because its not about how ini Entry pass to `Compiler`. Issue #1604 - more wider then just fix PHP code as Compiler pre-processes. This issue requires some changes in C-code generation. So, if you will test your code changes, you can see that duplicates of namespaces are still present in `phpinfo()` as in CLI output and as in HTML output.

Reply:
The reason for the #1604 duplication is that namespace and name use the same word. The correct treatment is to add namespace.

Reply:
As I said... for this issue we need change C-code generation also. I tested fix such as you provided and saw that some variables aren`t show in `phpinfo()`.

For example: `/ext/test.c`
```c
PHP_INI_BEGIN()
    STD_PHP_INI_BOOLEAN("test.test.my_setting_1", "0", PHP_INI_ALL, OnUpdateBool, test.my_setting_1, zend_test_globals, test_globals)
    ...
PHP_INI_END()
```

Reply:
I don't understand what you mean, Ccustom prefix name,  should use `ini-entry`, Otherwise, namespace is prefixed.